packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -726,10 +726,15 @@ export interface FoldingRules {
     markers?: FoldingMarkers;
 }
 
+export interface OnEnterRuleRegExp {
+    pattern: string;
+    flags?: string;
+}
```

Comment:
Looking at [this file](https://github.com/microsoft/vscode/blob/d912e6c92514b3dd524fe6ff1332a06c5f1d5534/src/vs/workbench/contrib/codeEditor/browser/languageConfigurationExtensionPoint.ts#L22-L25) in VSCode, it looks like there are a number of language configuration options that refer to this interface; shall we update all of the ones we know of - IndentationRule and FoldingRule - here?

Reply:
Oh yes, I suppose so.

Reply:
OK, I've adjusted the plugin-protocol interfaces to match. They all rely on the same createRegex method, so I don't think any additional runtime changes are necessary to handle those.